### PR TITLE
Flink Catalog Service Update

### PR DESCRIPTION
**Pull request by tony-zang**
Some update about the flink catalog:
Rewrite the flinkcatalogimpl class
Add one table configuration - "flink_catalog_configuration"
Add three maven dependency in scaleph-engine-sql module
Add one dict class

### CODE REVIEW

File: scaleph-common/src/main/java/cn/sliew/scaleph/common/dict/flink/FlinkCatalogType.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package cn.sliew.scaleph.common.dict.flink;
```
Comment (kalencaya) [inline scaleph-common/src/main/java/cn/sliew/scaleph/common/dict/flink/FlinkCatalogType.java:1]: lack apache license header here

File: scaleph-common/src/main/java/cn/sliew/scaleph/common/dict/flink/FlinkCatalogType.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import cn.sliew.scaleph.common.dict.DictInstance;
+import com.baomidou.mybatisplus.annotation.EnumValue;
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Arrays;
+
```
Comment (kalencaya) [inline scaleph-common/src/main/java/cn/sliew/scaleph/common/dict/flink/FlinkCatalogType.java:8]: lake @JsonFormat(shape = JsonFormat.Shape.OBJECT), which take effect when jackson serialize DictInstance class and then return {"label": "foo", "value": "bar"}

File: tools/docker/mysql/init.d/scaleph-flink-mysql.sql [hunk 3]
```diff
@@ -146,4 +146,22 @@ CREATE TABLE flink_job_log
...
+CREATE TABLE flink_catalog_configuration
+(
+    id                BIGINT       NOT NULL AUTO_INCREMENT,
+    flink_catalog_type_id BIGINT       NOT NULL,
+    flink_catalog_type_name VARCHAR(64)       NOT NULL,
+    catalog_name VARCHAR(64)       NOT NULL,
+    default_database VARCHAR(64)       NOT NULL,
+    hive_conf_dir VARCHAR(64) default NULL,
+    base_url VARCHAR(64) default NULL,
+    username VARCHAR(64) default NULL,
```
Comment (kalencaya) [inline tools/docker/mysql/init.d/scaleph-flink-mysql.sql:161]: default null is useless for mysql itself default behavior, we don't need to add it for clean code

Comment (tony-zang) [reply tools/docker/mysql/init.d/scaleph-flink-mysql.sql:161]: sure, I will fix all problem and re-submit a pr later

**Pull request merged**